leader-election-impl/src/main/metrics/leader-election.yml

```diff
@@ -0,0 +1,31 @@
+options:
+  javaPackage: 'com.palantir.leader'
+
+namespaces:
+  leaderElectionService:
+    docs: Metrics helpful for TimeLock adjudication.
+    metrics:
+      gainedLeadership:
+        type: meter
+        docs: Number of times(per second) leadership was gained successully.
+      lostLeadership:
+        type: meter
+        docs: Number of times(per second) leadership was lost.
+      noQuorum:
+        type: meter
+        docs: Number of times(per second) there was no quorum while determining leadership status.
+      proposedLeadership:
+        type: meter
+        docs: Number of times(per second) leadership was proposed.
+      proposalFailure:
+        type: meter
+        docs: Number of times(per second) leadership could not be gained after proposal.
```

Comment:
We should standardise between `Number of times(per second)` and `Rate`. I think I slightly prefer the former - or maybe `Rate (frequency per second)`?